### DISCUSSION TITLE
lag makes the example app unusable

### DISCUSSION THREAD

**Discussion by euri10** [Textualize/textual#4014] [Q&A] [answered]
greetings, 1st time user and running the `python -m textual` leads to the app being very laggy, pretty much unusable being so slow, I made a video recording (cant link it here but let me know if you want it) where I just run the app like in the docs, then it starts scrolling down without me pressing any input, as you'll see I have hard time selecting the inputs, finally enters a and b in it, then use my mouse wheel, it scrolls down line by line very laggy, I have to exit it pressing ctrl+c

when I ctrl+c it lefts me garbage in my console

![20240113_0710_1703x1055_1705126250](https://github.com/Textualize/textual/assets/1104190/a319e20d-baed-49f4-8be4-4eab398c06db)

**Comment by euri10** [comment]
tested in my terminal and it works fine, there should be something then with the Pycharm terminal

**Comment by davep** [comment]
A good place to start would be to tell us more about how and where you're attempting to run the demo; when you first raised this as an issue the issue template will have asked you to run `textual diagnose`; if you're still looking for help with this that sort of information will be vital to helping.

**Reply by euri10** [reply]
this is the output of diagnose, I tried it a my Pycharm terminal



# Textual Diagnostics

## Versions

| Name    | Value  |
|---------|--------|
| Textual | 0.47.1 |
| Rich    | 13.7.0 |

## Python

| Name           | Value                                                                              |
|----------------|------------------------------------------------------------------------------------|
| Version        | 3.9.9                                                                              |
| Implementation | CPython                                                                            |
| Compiler       | GCC 12.2.0                                                                         |
| Executable     | /home/lotso/.cache/pypoetry/virtualenvs/litestar-recipes--zOplDFp-py3.9/bin/python |

## Operating System

| Name    | Value                                               |
|---------|-----------------------------------------------------|
| System  | Linux                                               |
| Release | 6.1.0-17-amd64                                      |
| Version | #1 SMP PREEMPT_DYNAMIC Debian 6.1.69-1 (2023-12-30) |

## Terminal

| Name                 | Value          |
|----------------------|----------------|
| Terminal Application | *Unknown*      |
| TERM                 | xterm-256color |
| COLORTERM            | *Not set*      |
| FORCE_COLOR          | *Not set*      |
| NO_COLOR             | *Not set*      |

## Rich Console options

| Name           | Value                |
|----------------|----------------------|
| size           | width=224, height=53 |
| legacy_windows | False                |
| min_width      | 1                    |
| max_width      | 224                  |
| is_terminal    | True                 |
| encoding       | utf-8                |
| max_height     | 53                   |
| justify        | None                 |
| overflow       | None                 |
| no_wrap        | False                |
| highlight      | None                 |
| markup         | None                 |
| height         | None                 |

**Reply by davep** [reply]
Ahh, I see, thanks. The PyCharm terminal isn't a very capable terminal, from what I've seen. you could test the Textual demo against other full-screen terminal applications in the PyCharm terminal to see how they perform.

**Reply by euri10** [reply]
ok fair enough that was already what I noticed, this however leads to the question of how do people develop textual apps then 
1. do I have to go back and forth from pycharm to a "powerful-enough" terminal ?
2. how to use step debugging then ?
3. does vs code provides a better dx experience ?

**Reply by TomJGooding** [reply, marked as answer]
I don't use PyCharm or VSCode myself, but from what I've seen Textual runs far better in the VSCode integrated terminal. 

But Textual is designed to be run in a terminal emuator, and to quote @davep in a similar discussion recently: 

> More generally, the terminals provided by some IDEs tend to be quite cut-down and not a great place to run terminal-based applications in general (think of it as trying to test a full-featured website with all the bells and whistles, in a text-only web browser, and expecting to see all the fancy bells and whistles).

All that being said, if you really want to stick with PyCharm [there's apparently a setting  you could try](https://github.com/Textualize/textual/discussions/628#discussioncomment-7373369) which might make Textual at least somewhat usable.

**Reply by euri10** [reply]
thanks for the links @TomJGooding , I definitely should have found this myself but looking at issues led nowhere, a small mention in the docs would have been awesome, provided I had read it of course :)

**Comment by euri10** [comment]
https://github.com/Textualize/textual/assets/1104190/169ce0bd-6b4d-4705-946a-63d801e86b7f

**Comment by euri10** [comment]
1st example from the tutorial doesnt work:


https://github.com/Textualize/textual/assets/1104190/ebaab62b-ce34-464a-bb60-bd932cf30de6
